scaleway/resource_iam_policy.go

```diff
@@ -161,7 +161,10 @@ func resourceScalewayIamPolicyRead(ctx context.Context, d *schema.ResourceData,
 	if pol.ApplicationID != nil {
 		_ = d.Set("application_id", flattenStringPtr(pol.ApplicationID))
 	}
-	_ = d.Set("no_principal", flattenBoolPtr(pol.NoPrincipal))
+
+	if _, ok := d.GetOk("no_principal"); ok {
+		_ = d.Set("no_principal", flattenBoolPtr(pol.NoPrincipal))
+	}
```

Comment:
GetOk would supposedly fails if `no_principal` is not set but we're refreshing state

Reply:
GetOk does not work with boolean, it should be removed